xbout/tests/test_against_collect.py

```diff
@@ -19,14 +21,22 @@ def test_single_file(self, tmpdir_factory):
         generated_ds.to_netcdf(str(test_dir.join("BOUT.dmp.0.nc")))
 
         var = 'n'
-        expected = collect(var, path=test_dir, xguards=True, yguards=False)
+        expected = old_collect(var, path=test_dir, xguards=True, yguards=False)
 
+        # Test against new standard - open_boutdataset
         ds = open_boutdataset(test_dir.join("BOUT.dmp.0.nc"))
         actual = ds[var].values
 
         assert expected.shape == actual.shape
         npt.assert_equal(actual, expected)
 
+        # Test against backwards compatible collect function
+        actual = new_collect(var, path=test_dir, xguards=True, yguards=False)
+
+        assert expected.shape == actual.shape
+        npt.assert_equal(actual, expected)
```

Comment:
The tests all look good to me, but I'm wondering about best practice: should tests of `open_boutdataset` and tests of `new_collect` should be separate tests? They're (somewhat) independent things, so separate tests seems sensible. On the other hand, there would be a fair bit of copy-paste to do that, and extra creation/deletion of test files, so the way you've implemented is simpler and more efficient. Maybe @ZedThree has an opinion?

Reply:
Yeah I was having the same debate with myself. The copy and paste would be extensive as its only the variable `actual` that changes. I did test whether the whole test would fail if one `assert` in each function fails and the other passes, this is the case. But if we don't mind having more tmp files created it would be best to have a seperate test class dedicated to the new collect function.

Reply:
I don't think you don't need to test `new_collect` against `open_boutdataset` each time. They are so similar (just a thin wrapper) that a couple of tests to check the differences (like the way slicing is done) is sufficient. `open_boutdataset` should be treated as completely trusted if it passes it's own tests.

Reply:
I may be misunderstanding you Tom, but the `new_collect` is tested against `old_collect` and `open_boutdataset` is also tested against `old_collect`. If both tests pass we assume that `open_boutdataset` and `new_collect` as equivalent. 

Are you suggesting the values contained in `actual` for both `open_boutdataset` and `new_collect` should be checked against each other also? To ensure the data contain is the same.

Reply:
I might be the one who has got confused.

That set of tests you've described as already there sound fine though. Especially as `open_boutdataset` is also tested against synthetic data, so if `old_collect` has a bug (not unlikely) then we still have the synthetic data tests as the `"real"` answer.

Reply:
Correct. I think the tests as they are quite robust. Should a bug come up later down the line, likely related to `old_collect` then we can re-examine.